packages/two-factor-connector/src/modules/sagas.js

```diff
@@ -0,0 +1,41 @@
+import {takeLatest, all, call, put, select} from 'redux-saga/effects'
+import {rest} from 'tocco-app-extensions'
+import _get from 'lodash/get'
+
+import * as actions from './actions'
+
+const twoFactorField = 'relTwo_step_login_status'
+const twoFactorActiveState = '1'
+
+export const usernameSelector = state => state.twoFactorConnector.username
+
+export function* connectLogin() {
+  const username = yield select(usernameSelector)
+  const principalsResponse = yield call(rest.requestSaga, `principals/${username}/two-factor`, {method: 'POST'})
+  const {secret, totpUri} = principalsResponse.body
+  yield put(actions.setSecret({text: secret, uri: totpUri}))
+}
+
+export function* initialize() {
+  yield call(loadPrincipal2FAInfo)
+}
+
+export function* loadPrincipal2FAInfo() {
+  const principalsResponse = yield call(rest.requestSaga, 'principals')
+  const username = principalsResponse.body.username
+  yield put(actions.setUserName(username))
+  const tql = `username == "${username}"`
```

Comment:
usually we use the "fuzzy is" operator for username conditions (to query case insensitively)

TQL "fuzzy is" operator: === (instead of ==)

Reply:
got point. thanks